File: Windows/OpenIoT.App/OpenIoT.Lib/Board/Protocol/Events/IOpenIoTProtocolEvents.cs
using Palitri.OpenIoT.Board.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Palitri.OpenIoT.Board.Protocol.Events
{
    public interface IOpenIoTProtocolEvents : IPropertyTransmissionProtocolEvents
    {
        void OnDevicePropertiesReceived(object sender, Dictionary<int, byte[]> properties);
        void OnDevicePropertiesSet(object sender, Dictionary<int, byte[]> properties);
        void OnDeviceUidReceived(object sender, Guid uid);
        void OnDeviceNameReceived(object sender, string name);
        void OnProjectUidReceived(object sender, Guid uid);
        void OnProjectNameReceived(object sender, string name);
        void OnProjectHashReceived(object sender, uint hash);
        void OnUserUidReceived(object sender, Guid uid);
        void OnFirmwareNameReceived(object sender, string name);
        void OnFirmwareVendorReceived(object sender, string name);
        void OnFirmwareVersionReceived(object sender, string name);
        void OnBoardNameReceived(object sender, string name);



        void OnSchemeLogicUploaded(object sender);
        void OnProgramLogicUploaded(object sender);
        void OnResetLogic(object sender);
        void OnReset(object sender);
        void OnCommandExecuted(object sender);
    }
}

File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/board/protocol/events/IOpenIoTProtocolEvents.java
package com.palitri.openiot.construction.framework.board.protocol.events;

import java.util.HashMap;
import java.util.UUID;

public interface IOpenIoTProtocolEvents extends IPropertyTransmissionProtocolEvents {
    void onDevicePropertiesReceived(Object sender, HashMap<Integer, byte[]> properties);
    void onDevicePropertiesSet(Object sender, HashMap<Integer, byte[]> properties);
    void onDeviceUidReceived(Object sender, UUID uid);
    void onDeviceNameReceived(Object sender, String name);
    void onProjectUidReceived(Object sender, UUID uid);
    void onProjectNameReceived(Object sender, String name);
    void onProjectHashReceived(Object sender, int hash);
    void onUserUidReceived(Object sender, UUID uid);
    void onFirmwareNameReceived(Object sender, String name);
    void onFirmwareVendorReceived(Object sender, String name);
    void onFirmwareVersionReceived(Object sender, String name);
    void onBoardNameReceived(Object sender, String name);

    void onSchemeLogicUploaded(Object sender);
    void onProgramLogicUploaded(Object sender);
    void onResetLogic(Object sender);
    void onReset(Object sender);
 }
